doc/design/memory_optimization.md

```diff
@@ -0,0 +1,217 @@
+# Memory Optimization
+
+
+## Problem
+
+In a lecture from Andrew Ng, he attributes the recent sucess of AI due to a combination of these:
+
+- availability of Big Data
+- supercomputing power to process this Big Data over very large neural networks
+- modern algorithms
+
+Following graph shows the details:
+
+![](images/deep_learning.png)
+
+Larger model usually brings better performance. However, GPU memory is certain limited. For example, the memory size of a GTX TITAN X is only 12GB. To train complex and large model, we have to take care of memory using. Besides, memory optimization is also necessary in both online/mobile inference. 
```

Comment:
model -> **models**

Reply:
It should be `Larger **models** usually **bring**`

Reply:
It should be `However, GPU memory is limited`. We can drop the **certain**

Reply:
Thanks to @abhinavarora , I will merge it first.
@luotao1 The graphs will polished in next PR.